Car-service/src/main/java/carrest/CarServlet.java

```diff
@@ -0,0 +1,96 @@
+package carrest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.List;
+@WebServlet("/cars/*")
+public class CarServlet extends HttpServlet {
+    private final CarService carService = new CarService();
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String str = req.getPathInfo();
+        if (str == null) {
+            resp.setContentType("application/json");
+            List<Car> all = carService.getAll();
+            String json = mapper.writeValueAsString(all);
+            resp.getWriter().println(json);
+            System.out.println(all);
+        } else {
+            String[] parts = str.split("/");
+            String param = parts[1];
+            int carId = Integer.parseInt(param);
+            req.setCharacterEncoding("UTF-8");
+            resp.setContentType("application/json; charset=UTF-8");
+            Car car = carService.get(carId);
+            String json = mapper.writeValueAsString(car);
+            resp.getWriter().println(json);
+            System.out.println(car);
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        StringBuilder reqBody = new StringBuilder();
+        try (BufferedReader reader = req.getReader()) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                reqBody.append(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        String jsonData = reqBody.toString();
+        Car car = mapper.readValue(jsonData, Car.class);
+        carService.save(car);
```

Comment:
In this implementation you have no hibernate and DB that will autogenerate ID for entity class. So, in dao or service you have to add mechanism to generating and adding id's to users.

It's not critical, but just FYI